stumpy/scrump.py

```diff
@@ -645,48 +735,53 @@ def update(self):
                 self._T_B_subseq_isconstant,
                 self._diags[start_idx:stop_idx],
                 self._ignore_trivial,
+                self._k,
             )
 
-            # Update matrix profile and indices
-            for i in range(self._P.shape[0]):
-                if self._P[i, 0] > P[i, 0]:
-                    self._P[i, 0] = P[i, 0]
-                    self._I[i, 0] = I[i, 0]
-                # left matrix profile and left matrix profile indices
-                if self._P[i, 1] > P[i, 1]:
-                    self._P[i, 1] = P[i, 1]
-                    self._I[i, 1] = I[i, 1]
-                # right matrix profile and right matrix profile indices
-                if self._P[i, 2] > P[i, 2]:
-                    self._P[i, 2] = P[i, 2]
-                    self._I[i, 2] = I[i, 2]
+            # Update (top-k) matrix profile and indices
+            core._merge_topk_PI(self._P, P, self._I, I)
+
+            # update left matrix profile and indices
+            cond = PL < self._PL
+            self._PL[:] = np.where(cond, PL, self._PL)
+            self._IL[:] = np.where(cond, IL, self._IL)
+
+            # update right matrix profile and indices
+            cond = PR < self._PR
+            self._PR[:] = np.where(cond, PR, self._PR)
+            self._IR[:] = np.where(cond, IR, self._IR)
 
             self._chunk_idx += 1
 
     @property
     def P_(self):
         """
-        Get the updated matrix profile
+        Get the updated (top-k) matrix profile. When k=1 (default), the first (and only)
+        column in this 2D array consists of the matrix profile. When k > 1, the output
+        has exactly k columns consist of the top-k matrix profile.
         """
-        return self._P[:, 0].astype(np.float64)
+        return self._P.astype(np.float64)
 
     @property
     def I_(self):
         """
-        Get the updated matrix profile indices
+        Get the updated (top-k) matrix profile indices. When k=1 (default), the
+        first (and only) column in this 2D array consists of the matrix profile
+        indices. When k > 1, the output has exactly k columns consist of the top-k
```

Comment:
@seanlaw 
Could you please check out the sentence again?  Did you mean "is consisting of" ?  I think the verb "consists of" should be okay here.

Reply:
it would need to be "and consists of" or ", consisting of"

Reply:
My English is not as good as yours... but I am guessing you are seeing the word "is" instead of "in". Because, "consist" is the ONLY verb in the sentence. Sorry for being annoying on this :)

Reply:
"the output has exactly k columns consist of the top-k" sounds incorrect grammatically. It should be "the output has exactly k columns AND consistS of the top-k".

P.S. Not annoying

Reply:
Oops! I was reading the other sentence just before that. Got it. I will fix it. Thanks!